miden-tx/src/lib.rs

```diff
@@ -38,8 +38,8 @@ pub use error::{
     TransactionVerifierError,
 };
 
-#[cfg(test)]
-mod tests;
+#[cfg(any(test, feature = "testing"))]
+pub mod testing;
```

Comment:
I wonder if we should keep both `tests` and `testing` modules. The thinking is that `testing` is providing some utilities for testing while `tests` contains tests and doesn't provide testing utilities.

Reply:
So, instead of renaming, add a new module? I have no preferences, both works for me. But we do need the split to satisfy clippy's lints.